crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -447,18 +447,18 @@ fn estimate_output_nodes_for_curr_node(
 }
 
 /// Try to extract a u64 value from the given argument, or return None on failure.
-fn extract_limit(value: Option<&Positioned<GqlValue>>, variables: &Variables) -> Option<u64> {
+fn extract_limit(value: Option<&Positioned<GqlValue>>, variables: &Variables) -> Option<u32> {
     if let GqlValue::Variable(var) = &value?.node {
         return match variables.get(var) {
-            Some(Value::Number(num)) => num.as_u64(),
+            Some(Value::Number(num)) => num.as_u64().map(|v| v as u32),
             _ => None,
         };
     }
 
     let GqlValue::Number(value) = &value?.node else {
         return None;
     };
-    value.as_u64()
+    value.as_u64().map(|v| v as u32)
```

Comment:
Took me a second to wrap my head around this but I think it's fine because u64 maps to Int which on graphQL is capped at u32, so it shouldn't be possible for a graphQL client to provide > u32

Reply:
The query limits checker needs a deeper overhaul in any case so your confusion is warranted -- this should be clearer!